lib/Models/Terria.ts

```diff
@@ -408,8 +411,32 @@ export default class Terria {
     }
   }
 
+  getModelIdByShareKey(shareKey: string): string | undefined {
+    return this.shareKeysMap.get(shareKey);
+  }
+
+  getModelByIdOrShareKey<T extends BaseModel>(
+    type: Class<T>,
+    id: string
+  ): T | undefined {
+    let model = this.getModelById(type, id);
+    if (model) {
+      return model;
+    } else {
+      const idFromShareKey = this.getModelIdByShareKey(id);
+      return idFromShareKey !== undefined
+        ? this.getModelById(type, idFromShareKey)
+        : undefined;
+    }
+  }
+
+  @action
+  addShareKey(id: string, shareKey: string) {
+    this.shareKeysMap.set(shareKey, id);
+  }
```

Comment:
These might be better as private functions

Reply:
i'm happy to leave this for now